src/views/datalist/data-list-directive.js

```diff
@@ -0,0 +1,312 @@
+          $scope.eventText = item.name + ' clicked\n' + $scope.eventText;
+        };
+        var handleDblClick = function (item, e) {
+          $scope.eventText = item.name + ' double clicked\n' + $scope.eventText;
+        };
+        var handleCheckBoxChange = function (item, selected, e) {
+          $scope.eventText = item.name + ' checked: ' + item.selected + '\n' + $scope.eventText;
+        };
+
+        var checkDisabledItem = function(item) {
+          return $scope.showDisabled && (item.name === "John Smith");
+        };
+
+        $scope.showDisabled = false;
+
+        $scope.config = {
+         selectItems: false,
+         multiSelect: false,
+         dblClick: false,
+         selectionMatchProp: 'name',
+         selectedItems: [],
+         checkDisabled: checkDisabledItem,
+         showSelectBox: true,
+         rowHeight: 36,
+         onSelect: handleSelect,
+         onSelectionChange: handleSelectionChange,
+         onCheckBoxChange: handleCheckBoxChange,
+         onClick: handleClick,
+          onDblClick: handleDblClick
+        };
+
+        $scope.items = [
+          {
+            name: "Fred Flintstone",
+            address: "20 Dinosaur Way",
+            city: "Bedrock",
+            state: "Washingstone"
+          },
+          {
+            name: "John Smith",
+            address: "415 East Main Street",
+            city: "Norfolk",
+            state: "Virginia"
+          },
+          {
+            name: "Frank Livingston",
+            address: "234 Elm Street",
+            city: "Pittsburgh",
+            state: "Pennsylvania"
+          },
+          {
+            name: "Judy Green",
+            address: "2 Apple Boulevard",
+            city: "Cincinatti",
+            state: "Ohio"
+          },
+          {
+            name: "Pat Thomas",
+            address: "50 Second Street",
+            city: "New York",
+            state: "New York"
+          },
+        ];
+      };
+    ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.views').directive('pfDataList', [
+  function () {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        config: '=',
+        items: '=',
+        eventId: '@id'
+      },
+      replace: true,
+      transclude: true,
+      templateUrl: 'views/datalist/data-list.html',
+      controller: ['$scope',
+        function ($scope) {
+          $scope.defaultConfig = {
+            selectItems: false,
+            multiSelect: false,
+            dblClick: false,
+            selectionMatchProp: 'uuid',
+            selectedItems: [],
+            checkDisabled: false,
+            showSelectBox: true,
+            rowHeight: 36,
+            onSelect: null,
+            onSelectionChange: null,
+            onCheckBoxChange: null,
+            onClick: null,
+            onDblClick: null
+          };
+
+          $scope.config = $.extend(true, angular.copy($scope.defaultConfig), $scope.config);
```

Comment:
This is done to setup the default for cases where config is not set. I should have put =? on config so that it is optional for those who want pure default behavior.